packages/engine/src/behave-graph/nodes/Profiles/Engine/Values/ComponentNodes.ts

```diff
@@ -82,3 +87,47 @@ export const deleteComponent = makeFlowNodeDefinition({
     commit('flow')
   }
 })
+
+export const setTag = makeFlowNodeDefinition({
+  typeName: 'engine/component/tag/set',
+  category: NodeCategory.Action,
+  label: 'set Tag',
+  in: {
+    flow: 'flow',
+    entity: 'entity',
+    tagName: 'string'
+  },
+  out: { flow: 'flow', entity: 'entity', tagName: 'string' },
+  initialState: undefined,
+  triggered: ({ read, write, commit, graph: { getDependency } }) => {
+    const entity = Number.parseInt(read('entity')) as Entity
+    const tagName = read<string>('tagName')
+    const tag = defineComponent({ name: `bg-tag.${tagName}` })
+    setComponent(entity, tag)
+    write('entity', entity)
+    write('tagName', tagName)
+    commit('flow')
+  }
+})
+
+export const removeTag = makeFlowNodeDefinition({
+  typeName: 'engine/component/tag/remove',
+  category: NodeCategory.Action,
+  label: 'remove Tag',
+  in: {
+    flow: 'flow',
+    entity: 'entity',
+    tagName: 'string'
+  },
+  out: { flow: 'flow', entity: 'entity' },
+  initialState: undefined,
+  triggered: ({ read, write, commit, graph: { getDependency } }) => {
+    const entity = Number.parseInt(read('entity')) as Entity
+    const tagName = `bg-tag.${read<string>('tagName')}`
+    const component = ComponentMap.get(tagName)!
+    Assert.mustBeDefined(component, `Component ${tagName} does not exist`)
```

Comment:
IMO this should just be a no-op. All we care about is the end result (the tag/component is not on the entity).

Reply:
so more of a create tag? can we destroy components?